### ISSUE TITLE
`consult-completion-in-region` and geiser REPL not working properly

### ISSUE THREAD

**Issue by 0x462e41** [minad/consult#663]
I tried using the `consult-completion-in-region` command to complete symbol names within geiser REPL (more specifically, geiser-guile), but I ran into a bug where Emacs freezes. Without customizing the `completion-in-region-function` variable, Emacs doesn't hangs, so it's not a geiser problem.

A minimal config to replicate this bug is:

``` lisp
;; <skipping straight bootstrap snippet>
(dolist (package '(consult geiser geiser-guile))
  (straight-use-package package))
(setq-default completion-in-region-function #'consult-completion-in-region)
```

The bug is as follows:

1. Open the geiser REPL (`M-x geiser-guile`);
2. Start typing a S-expression like `(let_`, where '_' is the position of the point;
3. Evoke the command bound to `C-M-i` (`completion-at-point`) and then `?` to open the default completion listing;
4. Exit the completion evoking `C-g` (`keyboard-quit`). Everything should be working just fine until now;
5. Now, evoke again `completion-at-point` and Emacs should hang at this point.

I tried to profile Emacs, but I couldn't figure out what is the root cause of this bug, unfortunately. Let me know if you need any other info.

Best regards.

PS: My Emacs version is 28.2.

**Comment by minad** [comment]
Note that consult-completion-in-region doesn't make sense if you use the default completion UI. This means your minimal example is invalid.

Anyway, can you reproduce the hangup problem with Vertico/Icomplete? If yes, you should report this to Geiser instead, since the hangup occurs in the backend and we cannot workaround the problem in the frontend unfortunately.

cc the Geiser maintainer @jaor

**Comment by 0x462e41** [comment]
Yes, it freezes with Vertico/Icomplete too. I ran into the bug using Vertico at first, so I reduced init.el to the max to try to find the cause. It seems to be related to Consult as it doesn't freeze when I don't use `consult-completion-in-region`, either with Vertico or with default completion. I do understand it makes no sense to use `consult-completion-in-region` with the default completion, this was just a controlled test where it excludes all other confounding variables. I will contact the Geiser maintainer to see what they think about it, though.
